lib/cn/csched_sp3_work.c

```diff
@@ -714,37 +881,48 @@ sp3_work(
      * action running and then must release the token before returning.
      */
     have_token = cn_node_comp_token_get(tn);
-    if (!have_token) {
-        (*wp)->cw_resched = tn->tn_isroot;
+    if (!have_token)
```

Comment:
Once again, the root node token should always be available (I had changed that logic when I introduced spill/split synchronization, but reverted it with ispill).